### PR TITLE
Adds option to cancel only upcoming invocation

### PR DESCRIPTION
**Pull request by stamoern**
Issue: https://github.com/niksy/throttle-debounce/issues/28

### CODE REVIEW

**Comment by niksy** [comment]
@stamoern could you please update codebase to latest version if this is still something you need?

**Comment by stamoern** [comment]
> @stamoern could you please update codebase to latest version if this is still something you need?

Done!

**Comment by niksy** [comment]
@stamoern great, thanks!

Could you please apply [suggestion](https://github.com/niksy/throttle-debounce/pull/54#issuecomment-1101278998) from @seiyab ?

Also, it would be great to update changelog and draft type changes for DefinitelyTyped.

**Comment by niksy** [comment]
@stamoern great?

Could you please apply [suggestion](https://github.com/niksy/throttle-debounce/pull/54#issuecomment-1101278998) from @seiyab ?

Also, it would be great to update changelog and draft type changes for DefinitelyTyped.

**Comment by stamoern** [comment]
@niksy cancel API is normalized and changelog is updated. 

I will make the PR to DefinitelyTyped and will link it here

**Comment by niksy** [comment]
@stamoern please check with @seiyab so we can merge both changes in new major version bump (because of https://github.com/niksy/throttle-debounce/pull/54).

File: throttle.js [hunk 1]
```diff
@@ -46,9 +46,9 @@ export default function (delay, callback, options) {
 	}
 
 	// Function to cancel next exec
-	function cancel() {
+	function cancel(options) {
```
Comment (niksy) [inline throttle.js:49]: @stamoern could you please implement same check as here: https://github.com/niksy/throttle-debounce/blob/203deec24d78abf50751a48c10d0ca7ab1896806/throttle.js#L25-L29

File: README.md [hunk 2]
```diff
@@ -213,7 +228,7 @@ support).
 
 <!-- prettier-ignore-start -->
 
-**Original module license:** Copyright (c) 2010 "Cowboy" Ben Alman (Dual licensed under the MIT and GPL licenses. http://benalman.com/about/license/)
```
Comment (niksy) [inline README.md]: @stamoern we need double space here for line break

**Comment by seiyab** [comment]
@stamoern Are you still here?
@niksy Or, do you release 5.0.0 first?

**Comment by niksy** [comment]
Waiting on @stamoern first.

File: throttle.js (hunk 1 above)
Comment (stamoern) [reply throttle.js:49]: Done

**Comment by niksy** [comment]
@stamoern thanks for applying the changes.

Can we update types? /cc @seiyab

**Comment by seiyab** [comment]
@stamoern 
If you are willing to make PR for new type definition, switch a branch from (or cherry-pick) https://github.com/seiyab/DefinitelyTyped/commit/a019b27f5423ec4bc5773f333df6ccaa9cdbea61 .

@niksy 
FYI:
AFAIK, CI for type definition for major or minor update will fail until the version will be released.
It is better to releasing new version until @stamoern submit the PR, I think.

**Comment by stamoern** [comment]
@seiyab take a look at PR https://github.com/seiyab/DefinitelyTyped/pull/2
You can merge my changes with yours and update types after the new version of package will be released

**Comment by seiyab** [comment]
@stamoern Thank you for writing types.
@niksy type definition is ready 👍

**Comment by niksy** [comment]
@stamoern @seiyab major version published. Thank you all!

**Comment by seiyab** [comment]
types are updated. https://github.com/DefinitelyTyped/DefinitelyTyped/pull/60166

**Comment by niksy** [comment]
Thanks all!

**Pull request merged**